### ISSUE TITLE
Infinite loop in image download for some wiki with 0 images

### ISSUE THREAD

**Issue by nemobis** [WikiTeam/wikiteam#205]
I have a handful such wikis. Example:

Analysing http://wiki.homeactionreplay.org/api.php
Loading config file...
Resuming previous dump process...
Title list was completed in the previous session
XML dump was completed in the previous session
Image list is incomplete. Reloading...
Retrieving image filenames
.......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

**Comment by nemobis** [comment]
(I thought I had already commented, sorry.) Thanks for the patch! The fix gives an "of course! can't be wrong" feeling, but the query continuation system changed so many times that I'm not sure what wikis to test on. Let's try the above ones to start with. :)

With --images only, I get no loops, sot his specific bug is fixed for those wikis. However the numbers below are not correct for http://www.ctpedia.es/w/index.php/Especial:Statistics , 0 files are found out of 2051. We're probably trying to use an API feature not available in 1.9.

```
$ python launcher.py issue205 
#########################################################################
# Downloading http://wiki.homeactionreplay.org/api.php
#########################################################################
Checking API... http://wiki.homeactionreplay.org/api.php
API is OK
Checking index.php... http://wiki.homeactionreplay.org/index.php
index.php is OK
#########################################################################
# Welcome to DumpGenerator 0.3.0-alpha by WikiTeam (GPL v3)                   #
# More info at: https://github.com/WikiTeam/wikiteam                    #
#########################################################################

#########################################################################
# Copyright (C) 2011-2014 WikiTeam                                      #
# This program is free software: you can redistribute it and/or modify  #
# it under the terms of the GNU General Public License as published by  #
# the Free Software Foundation, either version 3 of the License, or     #
# (at your option) any later version.                                   #
#                                                                       #
# This program is distributed in the hope that it will be useful,       #
# but WITHOUT ANY WARRANTY; without even the implied warranty of        #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
# GNU General Public License for more details.                          #
#                                                                       #
# You should have received a copy of the GNU General Public License     #
# along with this program.  If not, see <http://www.gnu.org/licenses/>. #
#########################################################################

Analysing http://wiki.homeactionreplay.org/api.php
Trying generating a new dump into a new directory...
Retrieving image filenames
..    Found 0 images
0 image names loaded
Image filenames and URLs saved at... wikihomeactionreplayorg-20150214-images.txt
Retrieving images from "start"
Creating "./wikihomeactionreplayorg-20150214-wikidump/images" directory
Downloaded 0 images
Downloading index.php (Main Page) as index.html
Downloading Special:Version with extensions and other related info
Downloading site info as siteinfo.json
---> Congratulations! Your dump is complete <---
If you found any bug, report a new issue here: https://github.com/WikiTeam/wikiteam/issues
If this is a public wiki, please, consider publishing this dump. Do it yourself as explained in https://github.com/WikiTeam/wikiteam/wiki/Tutorial#Publishing_the_dump or contact us at https://github.com/WikiTeam/wikiteam
Good luck! Bye!
tail: impossibile aprire "wikihomeactionreplayorg-20150214-wikidump/wikihomeactionreplayorg-20150214-history.xml" per la lettura: File o directory non esistente
No </mediawiki> tag found: dump failed, needs fixing; resume didn't work. Exiting.
#########################################################################
# Downloading http://www.ctpedia.es/w/api.php
#########################################################################
Checking API... http://www.ctpedia.es/w/api.php
API is OK
Checking index.php... http://www.ctpedia.es/w/index.php
index.php is OK
#########################################################################
# Welcome to DumpGenerator 0.3.0-alpha by WikiTeam (GPL v3)                   #
# More info at: https://github.com/WikiTeam/wikiteam                    #
#########################################################################

#########################################################################
# Copyright (C) 2011-2014 WikiTeam                                      #
# This program is free software: you can redistribute it and/or modify  #
# it under the terms of the GNU General Public License as published by  #
# the Free Software Foundation, either version 3 of the License, or     #
# (at your option) any later version.                                   #
#                                                                       #
# This program is distributed in the hope that it will be useful,       #
# but WITHOUT ANY WARRANTY; without even the implied warranty of        #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
# GNU General Public License for more details.                          #
#                                                                       #
# You should have received a copy of the GNU General Public License     #
# along with this program.  If not, see <http://www.gnu.org/licenses/>. #
#########################################################################

Analysing http://www.ctpedia.es/w/api.php
Trying generating a new dump into a new directory...
Retrieving image filenames
..    Found 0 images
0 image names loaded
Image filenames and URLs saved at... ctpediaes_w-20150214-images.txt
Retrieving images from "start"
Creating "./ctpediaes_w-20150214-wikidump/images" directory
Downloaded 0 images
Downloading index.php (Main Page) as index.html
Downloading Special:Version with extensions and other related info
Downloading site info as siteinfo.json
---> Congratulations! Your dump is complete <---
If you found any bug, report a new issue here: https://github.com/WikiTeam/wikiteam/issues
If this is a public wiki, please, consider publishing this dump. Do it yourself as explained in https://github.com/WikiTeam/wikiteam/wiki/Tutorial#Publishing_the_dump or contact us at https://github.com/WikiTeam/wikiteam
Good luck! Bye!
tail: impossibile aprire "ctpediaes_w-20150214-wikidump/ctpediaes_w-20150214-history.xml" per la lettura: File o directory non esistente
No </mediawiki> tag found: dump failed, needs fixing; resume didn't work. Exiting.
#########################################################################
# Downloading http://ateneodecordoba.com/api.php
#########################################################################
Checking API... http://ateneodecordoba.com/api.php
API is OK
Checking index.php... http://ateneodecordoba.com/index.php
index.php is OK
#########################################################################
# Welcome to DumpGenerator 0.3.0-alpha by WikiTeam (GPL v3)                   #
# More info at: https://github.com/WikiTeam/wikiteam                    #
#########################################################################

#########################################################################
# Copyright (C) 2011-2014 WikiTeam                                      #
# This program is free software: you can redistribute it and/or modify  #
# it under the terms of the GNU General Public License as published by  #
# the Free Software Foundation, either version 3 of the License, or     #
# (at your option) any later version.                                   #
#                                                                       #
# This program is distributed in the hope that it will be useful,       #
# but WITHOUT ANY WARRANTY; without even the implied warranty of        #
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
# GNU General Public License for more details.                          #
#                                                                       #
# You should have received a copy of the GNU General Public License     #
# along with this program.  If not, see <http://www.gnu.org/licenses/>. #
#########################################################################

Analysing http://ateneodecordoba.com/api.php
Trying generating a new dump into a new directory...
Retrieving image filenames
.......................................    Found 19236 images
19236 image names loaded
Image filenames and URLs saved at... ateneodecordobacom-20150214-images.txt
Retrieving images from "start"
Creating "./ateneodecordobacom-20150214-wikidump/images" directory
Traceback (most recent call last):
  File "./dumpgenerator.py", line 1905, in <module>
    main()
  File "./dumpgenerator.py", line 1897, in main
    createNewDump(config=config, other=other)
  File "./dumpgenerator.py", line 1481, in createNewDump
    session=other['session'])
  File "./dumpgenerator.py", line 994, in generateImageDump
    session=session)  # use Image: for backwards compatibility
  File "./dumpgenerator.py", line 411, in getXMLFileDesc
    return("".join([x for x in getXMLPage( config=config, title=title, verbose=False, session=session)]))
  File "./dumpgenerator.py", line 518, in getXMLPage
    raise PageMissingError(params['title'], xml)
__main__.PageMissingError: page 'Special:Export' not found
tail: impossibile aprire "ateneodecordobacom-20150214-wikidump/ateneodecordobacom-20150214-history.xml" per la lettura: File o directory non esistente
No </mediawiki> tag found: dump failed, needs fixing; resume didn't work. Exiting.
```

**Comment by nemobis** [comment]
Unrelated, but ugly:

> __main__.PageMissingError: page 'Special:Export' not found

http://ateneodecordoba.com/index.php/Especial:Exportar
